tests/e2e/testdata/cross_namespace.yaml

```diff
@@ -0,0 +1,157 @@
+# Copyright Envoy AI Gateway Authors
+# SPDX-License-Identifier: Apache-2.0
+# The full text of the Apache license is available in the LICENSE file at
+# the root of the repo.
+
+# Create namespaces for the test
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: gateway-ns
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: route-ns
+---
+# GatewayClass (cluster-scoped resource)
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: cross-namespace-gatewayclass
+spec:
+  controllerName: gateway.envoyproxy.io/gatewayclass-controller
+---
+# Gateway in gateway-ns namespace
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: cross-namespace-gateway
+  namespace: gateway-ns
+spec:
+  gatewayClassName: cross-namespace-gatewayclass
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+---
```

Comment:
can you add this configuration to like others? 

https://github.com/envoyproxy/ai-gateway/blob/6eaaf401e062ce74d84747128a9143d528415cb8/tests/e2e/testdata/testupstream.yaml#L24-L42